libp2p/src/main/kotlin/io/libp2p/transport/implementation/ConnectionOverNetty.kt

```diff
@@ -41,14 +43,26 @@ open class ConnectionOverNetty(
     override fun secureSession() = secureSession
     override fun transport() = transport
 
-    override fun localAddress(): Multiaddr =
-        toMultiaddr(nettyChannel.localAddress() as InetSocketAddress)
-    override fun remoteAddress(): Multiaddr =
-        toMultiaddr(nettyChannel.remoteAddress() as InetSocketAddress)
+    override fun localAddress(): Multiaddr {
+        if (nettyChannel is QuicChannel) {
+            return toMultiaddr(nettyChannel.localSocketAddress() as InetSocketAddress)
+        } else {
+            return toMultiaddr(nettyChannel.localAddress() as InetSocketAddress)
+        }
+    }
+    override fun remoteAddress(): Multiaddr {
+        if (nettyChannel is QuicChannel) {
+            return toMultiaddr(nettyChannel.remoteSocketAddress() as InetSocketAddress)
+        } else {
+            return toMultiaddr(nettyChannel.remoteAddress() as InetSocketAddress)
+        }
+    }
```

Comment:
This is definitely needs to be refactored, but I would do it in a followup PR

Reply:
Looks nice! Want to merge it into this branch (there are 2 other commits here)?

Reply:
I would better isolate it tbh.

Reply:
Oh, you mean merge this first? Sure.